### ISSUE TITLE
Error `Undefined property: stdClass::$error_handling` after installation of LizMap Web Client

### ISSUE THREAD

**Issue by Nikeou** [3liz/lizmap-web-client#1013]
Hello,

I do an installation with the installer which seems correct

_Installation start..
[notice] Installation starts for the entry point index
All modules dependencies are ok
Module jelix installed
Module jacl2 installed
Module jacl2db installed
Module jacl2db_admin installed
Module jauthdb_admin installed
Module master_admin installed
Module jcommunity installed
Module admin installed
Module lizmap installed
Module view installed
Module dataviz installed
Module dynamicLayers installed
Module proj4php installed
All modules are installed or upgraded for the entry point index
[notice] Installation starts for the entry point admin
All modules dependencies are ok
Module jelix installed
Module jacl2 installed
Module jacl2db installed
Module jacl2db_admin installed
Module jauthdb_admin installed
Module master_admin installed
Module jcommunity installed
Module admin installed
Module lizmap installed
Module view installed
Module dataviz installed
Module dynamicLayers installed
Module proj4php installed
All modules are installed or upgraded for the entry point admin
[notice] Installation starts for the entry point script
All modules dependencies are ok
Module jelix installed
Module jacl2 installed
Module jacl2db installed
Module jacl2db_admin installed
Module jauthdb_admin installed
Module master_admin installed
Module jcommunity installed
Module admin installed
Module lizmap installed
Module view installed
Module dataviz installed
Module dynamicLayers installed
Module proj4php installed
All modules are installed or upgraded for the entry point script
Installation ended._ 

I can go a first time to http://localhost/index.php and see the Lizmap application home page with the demo project Montpellier - Transport.
But If I want to display the map, I have this error ::
"Notice: Undefined property: stdClass::$error_handling in C:\xampp\lizmap\prod_3.2.0\lib\jelix\core\jCoordinator.class.php on line 351"

If I refresh the index.php page, I always have this error... Connexion to the admin panel returns the same error too.

Can you please help me to understand why.

Thank's in advance.

**Comment by Nikeou** [comment]
Details about the installation environment ::
- Windows 10
- XAMPP installation of Apache 2.4.37
- PHP 7.2.12
- Lizmap Web Client 3.2.0

**Comment by abdelandre** [comment]
hello,
i have the same problem with : 
windows server 2008 R2
apache 2.4.37
php 5.6.38
client 3.2.0 and Master
"mod_fcgid: stderr: PHP Notice:  Undefined property: stdClass::$error_handling in E:\\LIZ\\lizmap\\prod\\master\\lib\\jelix\\core\\jCoordinator.class.php on line 351"
as soon as launch the map or the login admin page

**Comment by laurentj** [comment]
@Nikeou @abdelandre Could you zip the lizmap/var/config folder (without the profiles.ini.php file because I don't want to see your passwords :) ) and the temp/lizmap folder. And send them to my email dev@ljouanneau.com please?

I really don't see why it happens and I need more information like your lizmap configuration files. Thank you :-)

**Comment by abdelandre** [comment]
Hello Laurent,
Here it is
For info we are using another port : 83 (another apache in prod for ol4 “handmade” mapping in 80)
Thank you 😊
[cid:image001.png@01D010A4.C1089330]

[cid:image002.png@01D010A4.C1089330]<http://www.epfl.fr/>

Arnaud Buisson-Delandre
Responsable du bureau SI et géomatique
Ligne Directe : 03 83 80 61 44
Se rendre à l'Établissement Public Foncier de Lorraine<http://maps.google.fr/maps?f=q&hl=fr&q=rue+robert+blum+pont+%C3%A0+mousson&sll=47.15984,2.988281&sspn=18.777218,81.738281&ie=UTF8&z=15&ll=48.914124,6.064796&spn=0.008856,0.053988&om=1&iwloc=addr>
 [cid:image004.jpg@01D010A6.13A6D2E0]

Afin de contribuer à la protection de l'environnement, merci de n'imprimer ce courriel que si nécessaire




De : Laurent Jouanneau <notifications@github.com>
Envoyé : jeudi 22 novembre 2018 17:02
À : 3liz/lizmap-web-client <lizmap-web-client@noreply.github.com>
Cc : BUISSON-DELANDRE Arnaud <Arnaud.BUISSON-DELANDRE@epfl.fr>; Mention <mention@noreply.github.com>
Objet : Re: [3liz/lizmap-web-client] Error `Undefined property: stdClass::$error_handling` after installation of LizMap Web Client (#1013)


@Nikeou<https://github.com/Nikeou> @abdelandre<https://github.com/abdelandre> Could you zip the lizmap/var/config folder (without the profiles.ini.php file because I don't want to see your passwords :) ) and the temp/lizmap folder. And send them to my email dev@ljouanneau.com<mailto:dev@ljouanneau.com> please?

I really don't see why it happens and I need more information like your lizmap configuration files. Thank you :-)

**Comment by laurentj** [comment]
To investigate further, please modify files index.php and admin.php into lizmap/www : replace the line

```
jApp::setCoord(new jCoordinator());
```

by
```
jApp::setCoord(new jCoordinator('', false));
```

It deactivate the error manager of Lizmap, and then all PHP errors will be displayed by PHP  into the web page, or into your apache log files.

To display errors on the web page, be sure you have `display_errors=on` in your php.ini file of your server. (and you should restart php-fpm/php-cgi or apache)

Then please tell me errors that appear.

**Comment by laurentj** [comment]
Sorry, i made a mistake.

Keep the line `jApp::setCoord(new jCoordinator());` and replace the line

```
jApp::loadConfig('admin/config.ini.php');
// or
jApp::loadConfig('index/config.ini.php');
```

by

```
jApp::loadConfig('admin/config.ini.php', false);
// or
jApp::loadConfig('index/config.ini.php', false);
```

**Comment by laurentj** [comment]
Ok, I found the issue.

@rldhont added a new language recently, for translations : Norwegian. ( commit 697ef8ed7eb1e0b0430908e11cbd9f5fe364a759). Its code is `no_NO`. And the framework stores all available languages in a generated configuration file as `lang code = locale code`, and so for Norwegian, it stores `no=no_NO`.

Unfortunately, `no` is a reserved word in ini file. I known it for option values, but I just discover it is the case also for option name ! And so PHP translate `no` as a boolean value, and of course, it doesn't like it because this value is used as an option name in the ini file...

@Nikeou @abdelandre @jancelin to do a quick fix, the solution is to remove the `no_NO` from the config parameter `availableLocales` (you can see it in to the mainconfig.ini.php).  So add into your localconfig.ini.php (at the top):

```
availableLocales="cs_CZ,de_DE,el_GR,en_US,es_ES,eu_ES,fi_FI,fr_FR,gl_ES,hu_HU,it_IT,nl_NL,pl_PL,pt_BR,pt_PT,ro_RO,ru_RU,sl_SL,sv_SE"
```

You can also remove all language code you don't want to support in your lizmap installation. Keep at least `en_US` and your language code, so you can setup it to : 

```
availableLocales="en_US,fr_FR"
```

@rldhont I'll remove the `no_NO` language code until I will find a solution into Jelix.

**Comment by Nikeou** [comment]
Thank's a lot, It works !!

**Comment by laurentj** [comment]
Not closed yet, I have to fix the issue jelix/jelix/issues#328 into the jelix framework, so we could use the `no_NO` language

**Comment by nboisteault** [comment]
@laurentj can we close ths issue ?
